src/main/java/ru/yandex/clickhouse/response/ClickHouseResultSet.java

```diff
@@ -74,13 +78,23 @@
     // to the fact that rs.next() returned false.
     private boolean lastReached = false;
 
-    public ClickHouseResultSet(InputStream is, int bufferSize, String db, String table, boolean usesWithTotals, ClickHouseStatement statement, TimeZone timezone, ClickHouseProperties properties) throws IOException {
+    public ClickHouseResultSet(InputStream is, int bufferSize, String db, String table,
+        boolean usesWithTotals, ClickHouseStatement statement, TimeZone timeZone,
+        ClickHouseProperties properties) throws IOException
+    {
         this.db = db;
         this.table = table;
         this.statement = statement;
         this.properties = properties;
         this.usesWithTotals = usesWithTotals;
-        initTimeZone(timezone);
+        this.dateTimeTimeZone = timeZone;
+        this.dateTimeZone = properties.isUseServerTimeZoneForDates()
+            ? timeZone
+            : TimeZone.getDefault();
+        dateTimeFormat = new SimpleDateFormat(DATE_TIME_PATTERN);
+        dateTimeFormat.setTimeZone(dateTimeTimeZone);
+        dateFormat = new SimpleDateFormat(DATE_PATTERN);
+        dateFormat.setTimeZone(dateTimeZone);
```

Comment:
Let's move this fields initialization back to field declaration

Reply:
This is inconsistent with the other fields, but OK...